src/ol/Object.js

```diff
@@ -94,10 +94,10 @@ class BaseObject extends Observable {
     getUid(this);
 
     /**
+     * @name values_
      * @private
      * @type {!Object<string, *>}
      */
-    this.values_ = {};
```

Comment:
Rather than adding `@name`, this would be more consistent with the rest of the library:
```js
/**
 * @private
 * @type {!Object<string, *>}
 */
this.values_ = null;
```

Reply:
When I add the declaration `this.listeners_;` in the [Target class](/michalzielanski/openlayers/blob/517c9e6b52e1f88ad74bf9b998f7d4d75094a34a/src/ol/events/Target.js#L68) an _"TS2339: Property 'indexOf' does not exist on type '{}'"_ error appears on the line:
```js
    if (listeners.indexOf(listener) === -1) {
```
Adding types in the comments:
```js
let listeners = this.listeners_ || (this.listeners_ = /** @type {Object<string, Array<import("../events.js").Listener>>} */ {});
listeners = listeners[type] || (listeners[type] = /** @type {Array<import("../events.js").Listener>} */ []);
```
does not solve the problem. Is there a better solution than `@ts-ignore`?

Reply:
There is:
```js
/**
 * @private
 * @type {Object<string, *>}
 */
this.values_ = null;

Reply:
I have it this way (no `!` sign). The whole code looks like this:
```js
  constructor(opt_target) {
    // (...)

    /**
     * @private
     * @type {Object<string, Array<import("../events.js").Listener>>|undefined}
     */
    this.listeners_;
  }

  addEventListener(type, listener) {
    if (!type || !listener) {
      return;
    }
    let listeners =
      this.listeners_ ||
      (this.listeners_ = /** @type {Object<string, Array<import("../events.js").Listener>>} */ {});
    listeners =
      listeners[type] ||
      (listeners[
        type
      ] = /** @type {Array<import("../events.js").Listener>} */ []);
    if (listeners.indexOf(listener) === -1) {
      listeners.push(listener);
    }
  }
```
but the error occurs.


I also tried:
```js
/**
 * @private
 * @type {Object<string, Array<import("../events.js").Listener>>} // wo undefined
 */
this.listeners_;

/**
 * @private
 * @type {Object<string, Array<import("../events.js").Listener>>}
 */
this.listeners_ = null; // null assignment

/**
 * @private
 * @type {?Object<string, Array<import("../events.js").Listener>>} // question mark
 */
this.listeners_ = null;
```
and `@type {!` here:
```js
let listeners = this.listeners_ || (this.listeners_ = /** @type {!Object<string, Array<import("../events.js").Listener>>} */ {});
listeners = listeners[type] || (listeners[type] = /** @type {!Array<import("../events.js").Listener>} */ []);
```
Nothing helps.
All that remains is to use `@ts-ignore`?

Reply:
TypeScript does not like re-assigning a variable to a value with a different type. I took the liberty to make a change that fixes the type error, in addition to removing all `@name` annotations.